### PR TITLE
task is done

### PR DESCRIPTION
**Pull request by yermilov-m**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,64 @@
...
     @Override
     public int size() {
-        return -1;
+        int s = 0;
+        for (Object value : values) {
+            if (value != null) {
+                s++;
+            }
+        }
+        return s;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:41]:
```suggestion
        return size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,64 @@
...
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private int indexOfKey(K key) {
+        int index = -1;
+        for (int i = 0; i < keys.length; i++) {
+            if ((key == null && keys[i] == null) || key != null && key.equals(keys[i])) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:60]:
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,64 @@
...
+            } else if (k.equals(key)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private int indexOfKey(K key) {
+        int index = -1;
+        for (int i = 0; i < keys.length; i++) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:59]:
```suggestion
        for (int i = 0; i < size; i++) {
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,64 @@
...
+        int s = 0;
+        for (Object value : values) {
+            if (value != null) {
+                s++;
+            }
+        }
+        return s;
+    }
+
+    private boolean isKeyAvailable(K key) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:44]: delete this method.
You can call indexOfKey() and if it is -1 - this is new key

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,64 @@
...
+    }
+
     @Override
     public void put(K key, V value) {
+        if (isKeyAvailable(key)) {
+            keys[size] = key;
+            values[size] = value;
+            size = size();
+        } else {
+            values[indexOfKey(key)] = value;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:23]: change **size** here

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -3,17 +3,49 @@
...
+        if (index < 0) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            if (values[index] == null) {
+                size++;
+            }
+            values[index] = value;
+        }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:28]:
```suggestion
        if (index < 0) {
            keys[size] = key;
            values[size] = value;
            size++;
            return;
        }
        values[index] = value;
```

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**